TrafficCapture/captureOffloader/src/main/java/org/opensearch/migrations/trafficcapture/StreamChannelConnectionCaptureSerializer.java

```diff
@@ -313,6 +313,7 @@ private void addDataMessage(int captureFieldNumber, int dataFieldNumber, Instant
             // COS checked for unbounded limit above
             int availableCOSSpace = getOrCreateCodedOutputStreamHolder().getOutputStreamSpaceLeft();
             int chunkBytes = messageAndOverheadBytesLeft > availableCOSSpace ? availableCOSSpace - trafficStreamOverhead : byteBuffer.limit() - byteBuffer.position();
+            // Need to use slice over duplicate as we depend on limits
```

Comment:
I couldn't change these to duplicate as it seems the ByteBuffer::limit() logic relies on marks which are not duplicated with duplicate.

Reply:
I don’t fully understand this comment
I would rather the code read `slice().limit(chunkBytes).slice()`.
However, that feels like it could be accomplished other ways too - why do we need 2 slices?  Is this a spot where slice(rIdx, rIdx+chunkBytes) could make more sense?

Reply:
unfortunately, these are ByteBufers and not netty ByteBufs and so we don't have slice(rIdx..). I've since refactored this function to be more explicit about behavior and removing unnecessary slicing.